src/Sentry.NLog/SentryTarget.cs

```diff
@@ -0,0 +1,249 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+using NLog;
+using NLog.Config;
+using NLog.Targets;
+
+using Sentry.Extensibility;
+using Sentry.Infrastructure;
+using Sentry.Protocol;
+using Sentry.Reflection;
+
+namespace Sentry.NLog
+{
+    [Target("Sentry")]
+    public sealed class SentryTarget : TargetWithContext
+    {
+        private readonly Func<IHub> _hubAccessor;
+        private readonly ISystemClock _clock;
+        private IDisposable _sdkDisposable;
+
+        internal static readonly (string Name, string Version) NameAndVersion = typeof(SentryTarget).Assembly.GetNameAndVersion();
+
+        private static readonly string ProtocolPackageName = "nuget:" + NameAndVersion.Name;
+
+        public SentryTarget() : this(new SentryNLogOptions())
+        {
+        }
+
+        public SentryTarget(SentryNLogOptions options)
+            : this(
+                options,
+                () => HubAdapter.Instance,
+                sdkInstance: null,
+                SystemClock.Clock)
+        {
+        }
+
+        internal SentryTarget(SentryNLogOptions options, Func<IHub> hubAccessor, IDisposable sdkInstance, ISystemClock clock)
+        {
+            Debug.Assert(options != null);
+            Debug.Assert(hubAccessor != null);
+            Debug.Assert(clock != null);
+
+            // Overrides default layout. Still will be explicitly overwritten if manually configured in the
+            // NLog.config file.
+            Layout = "${message}";
+
+            Options = options;
+            _hubAccessor = hubAccessor;
+            _clock = clock;
+
+            if (sdkInstance != null)
+            {
+                _sdkDisposable = sdkInstance;
+            }
+        }
+
+        public SentryNLogOptions Options { get; }
+
+        #region Xml property setters
+
+        /// <summary>
+        /// Add any desired additional tags that will be sent with every message.
+        /// </summary>
+        [ArrayParameter(typeof(TargetPropertyWithContext), "tag")]
+        public IList<TargetPropertyWithContext> Tags => Options.Tags;
+
+        /// <summary>
+        /// The Data Source Name of a given project in Sentry.
+        /// </summary>
+        [RequiredParameter]
+        public string Dsn
+        {
+            get => Options.Dsn?.ToString();
+            set => Options.Dsn = new Dsn(value);
+        }
+
+        /// <summary>
+        /// Minimum log level for events to trigger a send to Sentry. Defaults to <see cref="LogLevel.Error" />.
+        /// </summary>
+        public string MinimumEventLevel
```

Comment:
Little confused about this property. Usually one uses the NLog-Logging-Rules for filtering of LogEvent-Level. Why also here? (Can understand the MinimumBreadcrumbLevel because that is a little special).

Reply:
Basically, as I understand it (and anyone please correct me if I'm wrong), in order to even run the `Write` method on the target, the target must be enabled for the events log level. So in order to still get the breadcrumb data on events that aren't considered "logged" to sentry, the `NLog` logging rules should specify the lowest possible level, which would then allow for configuring based on sentry's needs. Again, if I'm wrong on that I'll gladly switch it up. This decision was mainly based on the fact that they implemented it this way in the Serilog implementation.

Reply:
That sounds right. As you said, it's what was done in the Serilog integration.